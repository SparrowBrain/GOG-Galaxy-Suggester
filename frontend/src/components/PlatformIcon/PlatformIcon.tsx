import React from 'react';
import SvgIcon from '@material-ui/core/SvgIcon';
import Tooltip from '@material-ui/core/Tooltip';

import { platformIconProps } from './PlatformIcon.types';

import { ReactComponent as BattleNetIcon } from './icons/battlenet.svg'
import { ReactComponent as BethesdaNetIcon } from './icons/bethesda.svg'
import { ReactComponent as EpicIcon } from './icons/epic.svg'
import { ReactComponent as GogIcon } from './icons/gog.svg'
import { ReactComponent as HumbleBundleIcon } from './icons/humble_bundle.svg'
import { ReactComponent as OriginIcon } from './icons/origin.svg'
import { ReactComponent as ParadoxIcon } from './icons/paradox.svg'
import { ReactComponent as PlayStationIcon } from './icons/playstation.svg'
import { ReactComponent as SteamIcon } from './icons/steam.svg'
import { ReactComponent as UplayIcon } from './icons/uplay.svg'
import { ReactComponent as XboxIcon } from './icons/xbox.svg'
import { ReactComponent as DefaultIcon } from './icons/default.svg'

function PlatformIcon({ platform }: platformIconProps) {
    const icon = pickIcon(platform);

    return (
        <Tooltip title={platform} >
            <SvgIcon component={icon.icon} viewBox={`0 0 ${icon.size ?? 24} ${icon.size ?? 24}`} />
        </Tooltip>
    );
}

function pickIcon(platform: string) {
    switch (platform) {
        case 'battlenet':
        case 'Battle.net':
            return { icon: BattleNetIcon };
        case 'bethesda':
        case 'Bethesda.net':
            return { icon: BethesdaNetIcon };
        case 'epic':
        case 'Epic Games Store':
            return { icon: EpicIcon, size: 32 };
        case 'GOG':
            return { icon: GogIcon };
        case 'humble':
        case 'Humble Bundle':
            return { icon: HumbleBundleIcon };
        case 'origin':
        case 'Origin':
            return { icon: OriginIcon, size: 32 };
        case 'paradox':
        case 'Paradox Plaza':
            return { icon: ParadoxIcon };
        case 'psn':
        case 'PlayStation Network':
            return { icon: PlayStationIcon };
        case 'steam':
        case 'Steam':
            return { icon: SteamIcon };
        case 'uplay':
        case 'Uplay':
            return { icon: UplayIcon };
        case 'xboxone':
        case 'Xbox Live':
            return { icon: XboxIcon };
        default:
            return { icon: DefaultIcon };
    }
}

export default PlatformIcon;