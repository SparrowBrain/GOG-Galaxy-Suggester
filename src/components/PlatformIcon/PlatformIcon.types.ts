export interface platformIconProps {
    platform: string,
}