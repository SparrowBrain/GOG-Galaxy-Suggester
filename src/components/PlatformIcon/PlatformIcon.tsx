import React from 'react';
import SvgIcon from '@material-ui/core/SvgIcon';
import Tooltip from '@material-ui/core/Tooltip';

import { platformIconProps } from './PlatformIcon.types';

import { ReactComponent as BattleNetIcon } from './icons/battlenet.svg'
import { ReactComponent as BethesdaNetIcon } from './icons/bethesda.svg'
import { ReactComponent as EpicIcon } from './icons/epic.svg'
import { ReactComponent as GogIcon } from './icons/gog.svg'
import { ReactComponent as HumbleBundleIcon } from './icons/humble_bundle.svg'
import { ReactComponent as OriginIcon } from './icons/origin.svg'
import { ReactComponent as ParadoxIcon } from './icons/paradox.svg'
import { ReactComponent as PlayStationIcon } from './icons/playstation.svg'
import { ReactComponent as SteamIcon } from './icons/steam.svg'
import { ReactComponent as UplayIcon } from './icons/uplay.svg'
import { ReactComponent as XboxIcon } from './icons/xbox.svg'
import { ReactComponent as DefaultIcon } from './icons/default.svg'
import { IconButton } from '@material-ui/core';
import platformFromReleaseKey from '../../utils/platformFromReleaseKey';
import readablePlatformName from '../../utils/readablePlatformName';

function PlatformIcon({ releaseKey }: platformIconProps) {
    const platform = readablePlatformName(platformFromReleaseKey(releaseKey));
    const icon = pickIcon(platform);

    const onClick = () => {
        var win = window.open(`goggalaxy://openGameView/${releaseKey}`, '_blank');
        win?.focus();
    }

    return (
        <Tooltip title={platform} >
            <IconButton onClick={onClick}>
                <SvgIcon component={icon.icon} viewBox={`0 0 ${icon.size ?? 24} ${icon.size ?? 24}`} />
            </IconButton>
        </Tooltip>
    );
}

function pickIcon(platform: string) {
    switch (platform) {
        case 'Battle.net':
            return { icon: BattleNetIcon };
        case 'Bethesda.net':
            return { icon: BethesdaNetIcon };
        case 'Epic Games Store':
            return { icon: EpicIcon, size: 32 };
        case 'GOG':
            return { icon: GogIcon };
        case 'Humble Bundle':
            return { icon: HumbleBundleIcon };
        case 'Origin':
            return { icon: OriginIcon, size: 32 };
        case 'Paradox Plaza':
            return { icon: ParadoxIcon };
        case 'PlayStation Network':
            return { icon: PlayStationIcon };
        case 'Steam':
            return { icon: SteamIcon };
        case 'Uplay':
            return { icon: UplayIcon };
        case 'Xbox Live':
            return { icon: XboxIcon };
        default:
            return { icon: DefaultIcon };
    }
}

export default PlatformIcon;