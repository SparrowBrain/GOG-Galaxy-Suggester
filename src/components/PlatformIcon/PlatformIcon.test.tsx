import React from 'react';
import { mount } from 'enzyme';

import PlatformIcon from './PlatformIcon';

describe('PlatformIcon should', () => {
    it('show Battle.net icon when platform is Battle.net', () => {
        const wrapper = mount(<PlatformIcon platform={'Battle.net'}></PlatformIcon>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('show Bethesda.net icon when platform is Bethesda.net', () => {
        const wrapper = mount(<PlatformIcon platform={'Bethesda.net'}></PlatformIcon>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('show Epic icon when platform is Epic Games Store', () => {
        const wrapper = mount(<PlatformIcon platform={'Epic Games Store'}></PlatformIcon>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('show gog icon when platform is GOG', () => {
        const wrapper = mount(<PlatformIcon platform={'GOG'}></PlatformIcon>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('show Humble Bundle icon when platform is Humble Bundle', () => {
        const wrapper = mount(<PlatformIcon platform={'Humble Bundle'}></PlatformIcon>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('show Origin icon when platform is Origin', () => {
        const wrapper = mount(<PlatformIcon platform={'Origin'}></PlatformIcon>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('show paradox icon when platform is paradox', () => {
        const wrapper = mount(<PlatformIcon platform={'paradox'}></PlatformIcon>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('show PlayStation icon when platform is PlayStation Network', () => {
        const wrapper = mount(<PlatformIcon platform={'PlayStation Network'}></PlatformIcon>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('show Steam icon when platform is Steam', () => {
        const wrapper = mount(<PlatformIcon platform={'Steam'}></PlatformIcon>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('show Uplay icon when platform is Uplay', () => {
        const wrapper = mount(<PlatformIcon platform={'Uplay'}></PlatformIcon>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('show Xbox icon when platform is Xbox Live', () => {
        const wrapper = mount(<PlatformIcon platform={'Xbox Live'}></PlatformIcon>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('show default gog galaxy icon when platform is unknown', () => {
        const wrapper = mount(<PlatformIcon platform={'some random test platform'}></PlatformIcon>);
        expect(wrapper.html()).toMatchSnapshot();
    });

});