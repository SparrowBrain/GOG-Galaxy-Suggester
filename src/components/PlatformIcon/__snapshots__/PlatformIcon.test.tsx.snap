// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformIcon should show Battle.net icon when platform is Battle.net 1`] = `"<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" title=\\"Battle.net\\">battlenet.svg</svg>"`;

exports[`PlatformIcon should show Bethesda.net icon when platform is Bethesda.net 1`] = `"<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" title=\\"Bethesda.net\\">bethesda.svg</svg>"`;

exports[`PlatformIcon should show Epic icon when platform is Epic Games Store 1`] = `"<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 32 32\\" aria-hidden=\\"true\\" title=\\"Epic Games Store\\">epic.svg</svg>"`;

exports[`PlatformIcon should show Humble Bundle icon when platform is Humble Bundle 1`] = `"<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" title=\\"Humble Bundle\\">humble_bundle.svg</svg>"`;

exports[`PlatformIcon should show Origin icon when platform is Origin 1`] = `"<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 32 32\\" aria-hidden=\\"true\\" title=\\"Origin\\">origin.svg</svg>"`;

exports[`PlatformIcon should show PlayStation icon when platform is PlayStation Network 1`] = `"<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" title=\\"PlayStation Network\\">playstation.svg</svg>"`;

exports[`PlatformIcon should show Steam icon when platform is Steam 1`] = `"<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" title=\\"Steam\\">steam.svg</svg>"`;

exports[`PlatformIcon should show Uplay icon when platform is Uplay 1`] = `"<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" title=\\"Uplay\\">uplay.svg</svg>"`;

exports[`PlatformIcon should show Xbox icon when platform is Xbox Live 1`] = `"<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" title=\\"Xbox Live\\">xbox.svg</svg>"`;

exports[`PlatformIcon should show default gog galaxy icon when platform is unknown 1`] = `"<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" title=\\"some random test platform\\">default.svg</svg>"`;

exports[`PlatformIcon should show gog icon when platform is GOG 1`] = `"<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" title=\\"GOG\\">gog.svg</svg>"`;

exports[`PlatformIcon should show paradox icon when platform is paradox 1`] = `"<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" title=\\"paradox\\">paradox.svg</svg>"`;
